worker/operations.py

```diff
@@ -143,48 +149,132 @@ def cosmos_extraction(document_id, filename, downloaded_document, force_run=Fals
+        result_endpoint_text = f"{extraction_json['result_endpoint']}/text"
+        equations_endpoint = (
+            f"{extraction_json['result_endpoint']}/extractions/equations"
+        )
+        figures_endpoint = f"{extraction_json['result_endpoint']}/extractions/figures"
+        tables_endpoint = f"{extraction_json['result_endpoint']}/extractions/tables"
 
         job_done = False
 
         for i in range(MAX_ITERATIONS):
             status = requests.get(status_endpoint)
             status_data = status.json()
-            logger.info("Polled status endpoint %s times:\n%s", i+1, status_data)
-            job_done = status_data['error'] or status_data['job_completed']
+            logger.info("Polled status endpoint %s times:\n%s", i + 1, status_data)
+            job_done = status_data["error"] or status_data["job_completed"]
             if job_done:
                 break
             time.sleep(POLLING_INTERVAL)
 
         if not job_done:
             logger.error("ERROR: Job not complete after %s seconds.", MAX_EXECTION_TIME)
             raise Exception(f"Job not complete after {MAX_EXECTION_TIME} seconds.")
-        elif status_data['error']:
+        elif status_data["error"]:
             logger.error("An unexpected error occurred: %s", {status_data["error"]})
-            raise Exception(f"An error occurred when processing in Cosmos: {status_data['error']}")
+            raise Exception(
+                f"An error occurred when processing in Cosmos: {status_data['error']}"
+            )
+
+        logger.info(result_endpoint_text)
+        result = requests.get(result_endpoint_text)
+
+        # Download the Cosmos extractions zipfile to a temporary directory
+        temp_dir = tempfile.mkdtemp()
+        zip_file = os.path.join(temp_dir, document_id + ".zip")
+        with open(zip_file, "wb") as writer:
+            writer.write(requests.get(result_endpoint).content)
+
+        # Extract zipfile to enable asset uploading
+        with zipfile.ZipFile(zip_file, "r") as zip_ref:
+            zip_ref.extractall(temp_dir)
+
+        # Assets requests
+        equations_resp = requests.get(equations_endpoint)
+        figures_resp = requests.get(figures_endpoint)
+        tables_resp = requests.get(tables_endpoint)
+        responses = {
+            "equation": equations_resp,
+            "figure": figures_resp,
+            "table": tables_resp,
+        }
+
+        # Checks status of responses, then adds their JSON content to
+        # an iterator object if they were successful.
+        assets_iterator = {}
+        for key, response in responses.items():
+            logger.info(response.status_code)
+            if response.status_code >= 300:
+                continue
+            else:
+                assets_iterator[key] = response.json()
+
+        assets = []
+        for key, value in assets_iterator.items():
+            for record in value:
+                path = record.get("img_pth")
```

Comment:
what if `img_pth` doesn't exist? do we need a check on that--e.g. `if not path` on the next line? I'm not sure 🤷‍♂️